Sources/Basics/ConcurrencyHelpers.swift

```diff
@@ -0,0 +1,66 @@
+/*
+ This source file is part of the Swift.org open source project
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+
+public struct ThreadSafeDictionary<Key, Value> where Key: Hashable {
```

Comment:
Naming question:  Would it be better to use a name other than "Dictionary" here since it doesn't implement all the methods of Dictionary, i.e. can't be used interchangeably with it?

Reply:
🚲 🏠 : `ThreadSafeKeyValueStore`?